ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs

```diff
@@ -405,7 +406,9 @@ instance ShelleyBasedEra era => BlockSupportsMetrics (ShelleyBlock era) where
          SelfIssued    -> IsSelfIssued
          NotSelfIssued -> IsNotSelfIssued
 
-instance ShelleyBasedEra era => RunNode (ShelleyBlock era)
+instance ( SerialiseNodeToClientConstraints (ShelleyBlock era)
```

Comment:
My initial expectation is that we shouldn't need to add `SerialiseNodeToClientConstraints` here.

Edit: Hmm. seems like you're using this to workaround module hierarchies/avoiding orphans/etc. Is that it? I'm not ecstatic about this approach, but I don't yet obviously see simple path to something cleaner.

Reply:
Yes, this is something I've struggled with a few times: I add some "unneeded" constraints because importing the relevant orphan instance would cause a module cycle. This is one of the reasons to dislike orphan instances. I also don't see an clear solution short of getting rid of the relevant orphan instances which sounds like a significant refactoring though.

Trying to move this instance into `Ouroboros.Consensus.Shelley.Ledger.Block` so that it itself is at least not an orphan is messy too. We require most of the other orphan instances in this module and can't import it without creating another module cycle. The existing orphans in a way force us to make more orphans :unamused: .

Reply:
Would you open a GitHub Issue in this repo, labeled `consensus` and `tech debt` about trying to wrangle these orphans? Or at least better document the method to the madness?

I agree it's likely a significant refactoring, but I think the state of the problem is better represented in your head right now than it has been for anyone, maybe ever. If you could write some of that insight down it could help us eventually approach that task 🙏

Reply:
Done #3302